src/graphql.tsx

```diff
@@ -320,18 +316,15 @@ export default function graphql(
       }
 
       setInitialProps() {
-        if (this.type === DocumentType.Mutation) return this.createWrappedMutation(this.props);
+        if (this.type === DocumentType.Mutation) {
+          return;
+        }
 
         // Create the observable but don't subscribe yet. The query won't
         // fire until we do.
         const opts: QueryOptions = this.calculateOptions(this.props);
 
-        if (opts.skip) {
-          this.data = assign({}, skippedQueryData) as any;
-        } else {
-          this.data = assign({}, defaultQueryData) as any;
-          this.createQuery(opts);
-        }
+        this.createQuery(opts);
```

Comment:
It looks like we now call createQuery unconditionally on setInitialProps (for non-mutations), even if the query is initially skipped. Isn't that wrong?